pkg/client/ocm/client.go

```diff
@@ -56,6 +56,8 @@ type Client interface {
 	GetQuotaCostsForProduct(organizationID, resourceName, product string) ([]*amsv1.QuotaCost, error)
 	// GetCurrentAccount returns the account information of the current authenticated user
 	GetCurrentAccount() (*amsv1.Account, error)
+	// GetSubscriptions returns the list of subscriptions that matches the given search criteria
+	GetSubscriptions(page, size int, search string) ([]*amsv1.Subscription, error)
```

Comment:
I think it might be more accurate to call it ListSubscriptions as that is how AMS calls it

Reply:
There is also a FindSubscriptions method that I think might have overlapping functionality. Do we really need both methods?

Reply:
Good spot. I believe we could enhance the `FindSubscriptions` to add page and pagesize and get rid of this method.
To avoid breaking the current code, we could use `functional options`:

```go
func Page(page int) FindSubscriptionsPageParams {
	return func(r *amsv1.SubscriptionsListRequest) {
		r.Page(page)
	}
}

func PageSize(pageSize int) FindSubscriptionsPageParams {
	return func(r *amsv1.SubscriptionsListRequest) {
		r.Size(pageSize)
	}
}

type FindSubscriptionsPageParams func(r *amsv1.SubscriptionsListRequest)

func (c client) FindSubscriptions(query string, params ...FindSubscriptionsPageParams) ([]*amsv1.Subscription, error) {
	req := c.connection.AccountsMgmt().V1().Subscriptions().List().Search(query)
	for _, param := range params {
		param(req)
	}

	r, err := req.Send()
	if err != nil {
		return nil, err
	}
	return r.Items().Slice(), nil
}
```

This way, we could call the method as:
```go
// no options
c.FindSubscriptions(query)

// specific page
c.FindSubscriptions(query, Page(3))

// specific page with page size
c.FindSubscriptions(query, Page(3), PageSize(50))
```

Reply:
I am not concerned about breaking the code. It should not be too difficult to refactor.

I am unsure about whether we need to go as far as using the functional options patterns for paging of the method.
I don't expect the set of paging options to be very high or unpredictable so in that case I tend to prefer to keep it simple and just use a typed struct with the options. More options can be added in the future is needed as attributes.
It will keep the signature easy, there will be less abstraction and indirections and if we need to add a new paging option (which won't be frequent) it should be as easy as adding a new attribute.
It would also avoid having a variadic argument which carry the implication that there cannot be other additional arguments after it.

As explained, for a case like this I would prefer to keep it simple and have a solution that contains a balance between simplicity and practicality.

Reply:
> I am not concerned about breaking the code. It should not be too difficult to refactor.

Yes, but we will need to pass the new parameter even when unnecessary.

> It would also avoid having a variadic argument which carry the implication that there cannot be other additional arguments after it.

True, but even `page, pagesize` are parameters passed typically as the first or last parameter, so passing other parameters after the `page` will be unusual.

Anyway, the implementation detail is debatable, and I'm open to any solution.

Reply:
I see your points and the first point is certainly an implication (as using functional options has other implications)

:+1: to your comment, I think both options are reasonable too. I would tend to simplicity in this specific case for the reasons outlined (there are other cases where using the functional options pattern is a very appropriate pattern) but I am open to whatever is chosen.

Reply:
@ziccardi @miguelsorianod the existing `FindSubscriptions` method is sufficient. For this use case, size was added to ensure only one item is returned but that's not really needed since the constructed query is unique enough to return one element.

Reply:
done, thanks for catching this one